tools/query/query.xml

```diff
@@ -0,0 +1,134 @@
+        <requirement type="package">pyarrow</requirement>
+        <requirement type="package">pytables</requirement>
+        <requirement type="package">pandas</requirement>
+        <requirement type="package">pandasql</requirement>
+        <requirement type="package" version="3.7">python</requirement>
+    </requirements>
+
+    <command><![CDATA[
+        python '$__tool_directory__/query.py'
+            $config
+    ]]></command>
+
+    <configfiles>
+        <configfile name="config"><![CDATA[
+            #def filetype(data):
+                #if $data.is_of_type('h5')
+                    #return 'hdf'
+                #elif $data.is_of_type('csv')
+                    #return 'csv'
+                #elif $data.is_of_type('tsv', 'tabular')
+                    #return 'tsv'
+                #elif $data.is_of_type('feather', 'arrow')
+                    #return 'feather'
+                #elif $data.is_of_type('parquet')
+                    #return 'parquet'
+                #elif $data.is_of_type('sqlite')
+                    #return 'sqlite'
+                #end if
+            #end def
+
+            #set $config = {
+                'tables': [
+                    {
+                        'path': str($table.file),
+                        'name': str($table.name) if $table.name else 't{}'.format(str($index)),
+                        'format': str($table.format) if $table.format else $filetype($table.file)
+                    }
+                    for $index, $table in enumerate($tables)
+                ],
+                'query': str($query),
+                'result': {
+                    'path': str($output),
+                    'name': str($result.name) if str($result.format) in ('h5', 'hdf', 'sqlite') and $result.name else None,
+                    'format': str($result.format)
+                }
+            }
+
+            #import json
+            #echo $json.dumps($config)
+        ]]></configfile>
+    </configfiles>
+
+    <inputs>
+        <repeat name="tables" title="Table">
+            <param name="file" type="data" format="csv,tsv,h5,feather,parquet,sqlite" label="File"/>
+            <param name="format" type="select" optional="true">
+                <label>Force the file format</label>
+                <help>Force a particular file format to get around formats unknown to Galaxy, ie. Parquet.</help>
+                <expand macro="formats"/>
+            </param>
+            <param name="name" type="text" optional="true">
+                <label>Table name</label>
+                <help>Choose an unique name for the table which then you may use in the SQL Query. By default tables will be named as: t0, t1, ...</help>
+            </param>
+        </repeat>
+        <param name="query" type="text" area="true">
+            <label>SQL Query</label>
+            <help>
+                Type an arbitrary SQL SELECT to perform on the input tables. The input tables may be referred by their given names.
```

Comment:
arbitrary SQL query could be scary, we need to check for security implications of giving such tool to users

Reply:
As long as `pandasql` can not be used to execute code it should be fine since the input is opened read only. The worst thing that I could imagine is that the output is messed up.

Reply:
Its also worth looking at https://github.com/galaxyproject/tools-iuc/blob/master/tools/gemini/gemini_macros.xml#L101 and Gemini in general as those tools also deal with sqlite.